### PR TITLE
Updating Enso Cloud integration - cheaper `Enso_Secret.create`, introducing `Enso_File.creation_time` and `last_modified_time`

### LINKED ISSUE #9047: Update Enso cloud integration to latest cloud updates
https://docs.google.com/document/d/1x8RUt3KkwyhlxGux7XUGfOdtFSAZV3fI9lSSqQ3XsXk/edit

- [x] `create_directory` returns org ID so we can update the tests
- [x] update URI paths for secrets endpoints (`update_secret`)
- [x] can we replace `exists` check on the secret with only cloud-side check? 
  - Yes, the endpoint gives us a clearer error, we need to parse it.
- ~~Try switching `Enso_File.list` to use `GET /directories/<id>/` endpoint instead of `list_assets` workaround~~ misunderstood
  - [x] Switch to `GET /directories/<id>` from `/directories?parent_id=<id>`
- Update `Enso_File` metadata tests:
   - [x] creation time and last modified time
   - ~~parent, path~~ - not enough functionality available
- ~~Update `Enso_File.read` to use presigned URLs if already available.~~
  - Not yet available as the feature was reverted. To be addressed in a separate PR.

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

- Closes #9047

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Enso_Cloud/Enso_File.enso [hunk 1]
```diff
@@ -344,15 +349,16 @@ Enso_Asset_Type.from (that:Text) = case that of
...
-get_s3_url_for_file file:Enso_File -> Text =
+   The returned URL may be valid only for a very short time, so it should not be
+   stored anywhere.
+get_download_url_for_file file:Enso_File -> Text =
     file_description = get_file_description file
-    path = file_description.get "path" if_missing=(Error.throw (Illegal_State.Error "Invalid JSON for an Enso_File (missing `path`): "+file_description.to_text))
+    ## TODO migrate to the pre-signed URLs once that is implemented
+       presigned_url = file_description |> get_required_field "url"
+    path = file_description |> get_required_field "file" |> get_required_field "path"
     path.replace "s3://production-enso-organizations-files/" "https://production-enso-organizations-files.s3.eu-west-1.amazonaws.com/"
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Enso_Cloud/Enso_File.enso:363]: Pre-signed URLs are not enabled yet as the feature was apparently rolled-back.

They can be enabled and will then be tested once it is available again, possibly in another PR.

File: test/Base_Tests/src/Network/Enso_Cloud/Enso_File_Spec.enso [hunk 2]
```diff
@@ -153,8 +149,16 @@ add_specs suite_builder setup:Cloud_Tests_Setup = setup.with_prepared_environmen
...
             nested_file.is_regular_file.should_be_true
             nested_file.size . should_equal nested_file.read_bytes.length
+            nested_file.creation_time . should_be_a Date_Time
+            nested_file.last_modified_time . should_be_a Date_Time
+
+        group_builder.specify "currently does not support metadata for directories" <|
+            # TODO this test should be 'reversed' and merged with above once the metadata is implemented
+            dir = Enso_File.root / "test-directory"
+            Test.expect_panic Unimplemented dir.creation_time
+            Test.expect_panic Unimplemented dir.last_modified_time
```
Comment (radeusgd) [inline test/Base_Tests/src/Network/Enso_Cloud/Enso_File_Spec.enso:159]: Related ticket: https://github.com/enso-org/cloud-v2/issues/911

File: tools/http-test-helper/src/main/java/org/enso/shttp/auth/HandlerWithTokenAuth.java [hunk 3]
```diff
@@ -31,7 +31,7 @@ protected void doHandle(HttpExchange exchange) throws IOException {
     String providedToken = authHeader.substring(prefix.length());
     boolean authorized = providedToken.equals(getSecretToken());
     if (!authorized) {
-      sendResponse(403, "Invalid token.", exchange);
+      sendResponse(401, "Invalid token.", exchange);
```
Comment (radeusgd) [inline tools/http-test-helper/src/main/java/org/enso/shttp/auth/HandlerWithTokenAuth.java:34]: This is updated for consistency with the real cloud behaviour.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Enso_Cloud/Utils.enso [hunk 4]
```diff
@@ -57,3 +75,44 @@ internal_cloud_project_directory = AuthenticationProvider.getCurrentWorkingDirec
...
+http_request_as_json (method : HTTP_Method) (url : URI) (body : Request_Body = Request_Body.Empty) (additional_headers : Vector = []) (error_handlers : Map Text (Any -> Any) = Map.empty) -> Any ! Enso_Cloud_Error =
+    response = http_request method url body additional_headers error_handlers
+    response.decode_as_json.catch Invalid_JSON error->
+        Error.throw (Enso_Cloud_Error.Invalid_Response_Payload error)
+
+## PRIVATE
+   Performs a standard request to the Enso Cloud API,
+   returning a raw response on success, but handling the error responses.
+
+   Custom error handles can be provided as a mapping from error codes
```
Comment (GregoryTravis) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Enso_Cloud/Utils.enso]:
```suggestion
   Custom error handlers can be provided as a mapping from error codes
```

**Review by GregoryTravis — APPROVED** [review]

**Review by jdunkerley — APPROVED** [review]

**Pull request merged**